lib/pbench/cli/server/user_update.py

```diff
@@ -0,0 +1,107 @@
+import click
+from pbench import BadConfig
+from pbench.server import PbenchServerConfig
+from pbench.cli.server import pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server.database.models.users import User
+from pbench.server.database.database import Database
+
+_NAME_ = "pbench-user-update"
+
+
+class UserUpdate:
+    def __init__(self, context):
+        self.context = context
+
+    def execute(self):
+        config = PbenchServerConfig(self.context.config)
+
+        logger = get_pbench_logger(_NAME_, config)
+
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access.
+        Database.init_db(config, logger)
+
+        # Check if the user has the right privileges
+        if user is None or not user.is_admin():
+            raise RuntimeError(f"Not authorized to update user {self.context.username}")
+
+        dict_to_update = {}
+        if self.context.username:
+            dict_to_update["username"] = self.context.username
+
+        if self.context.first_name:
+            dict_to_update["first_name"] = self.context.first_name
+
+        if self.context.last_name:
+            dict_to_update["last_name"] = self.context.last_name
+
+        if self.context.email:
+            dict_to_update["email"] = self.context.email
+
+        if self.context.role:
+            dict_to_update["role"] = self.context.role
+
+        # If the authorization passed update the user from the given username
+        User.update(dict_to_update)
+
+        click.echo(f"User {self.context.user_to_update} updated")
+
+
+@click.command()
```

Comment:
Nikhil, I'd like to ask a favor:  could you investigate creating a "command group" for the delete, update, and list commands?

Ultimately, we want to be able to type the `pbench` command at the shell prompt, and then type `user create` and/or `user delete` at the `pbench>` prompt (or even type `user` and receive a `pbench user>` prompt...).  In order to make that possible, we need to have the pbench commands grouped at appropriate levels.

I don't think that setting up these groups precludes having "stand-alone" tools (that is, I think we can still invoke this tool via `pbench-user-update`), but it will move us along toward the CLI implementation that we are trying to get to.  (I _think_ that all you have to do is replace the `@click.command()` with `@click.group(name='user_group')` or something similar.)